### ISSUE TITLE
Floating Menu: Revolving Keyboard Nav

### ISSUE THREAD

**Issue by BrittanyIRL** [swissspidy/web-stories-wp#11162]
## Feature Description

Release 1.19 introduced [floating menus ](https://github.com/GoogleForCreators/web-stories-wp/issues/10640) but under the time constraints, keyboard navigation was sacrificed to return to later. 

Initial work to get to the point where the floating menus themselves can have keyboard nav addressed was done in #10872  but the scope of that expanded, so in an effort to isolate changes the menu updates are here now. 

## Acceptance Criteria 

- The floating menus should have a [role of `toolbar` ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/toolbar_role).
- Floating menu toolbar has revolving keyboard nav, where pressing right on the last item returns you to the first, (only pressing enter on the more button and esc will navigate you out of the floating menu)

Please include tests that take into consideration the following (at the minimum): 
- Key into a floating menu + edit from a dropdown + return to element 
- Key into a floating menu + edit a text/number input + return to element 
- Key into a floating menu + edit from a popup (color picker) + return to element 
- Key through the whole floating menu and get back to the first focusable element in the menu.

**Comment by BrittanyIRL** [comment]
@felipebochehin87 there's quite a bit of detail in this PR description that will be useful for testing. 

Several known issues that aren't in scope here: 
- https://github.com/GoogleForCreators/web-stories-wp/issues/11312 (the outline on the color opacity (shape) encompasses the dash 
- https://github.com/GoogleForCreators/web-stories-wp/issues/11311 the focus outline when you first enter the floating menu via keyboard (`ctrl+option+p` isn't present even though that element _is_ focused (you can use it) 
- https://github.com/GoogleForCreators/web-stories-wp/issues/11003 Escaping the menu back to the element only works if you started by entering the canvas with the keyboard.

**Comment by felipebochehin87** [comment]
Tested against https://stories-qa-wordpress-amp.pantheonsite.io/wp-admin/, using PR #11273.

1. Added elements to the canvas
2. User Ctrl+Alt+P to access floating menu
3. User arrows, Enter and numeric keyboard to manipulate the options on floating menu
4. Pressed Esc and focus returned to the element

Looks good, except for the known issues we have, as mentioned in the PR:

- _"Eyedropper is still really weird to use with a keyboard, but that's not new"_
- _"One outstanding item is getting the canvas focused again if you didn't start out with just the keyboard"_
